### PR TITLE
[MIG] sale_automatic_workflow: Migration to 15.0

### PR DESCRIPTION
**Pull request by yankinmax**
PR to migrate sale_automatic_workflow to V15.
Was ported from V14 here:
- [x] https://github.com/OCA/sale-workflow/pull/1857

Depends on:

- [x] #1851

### CODE REVIEW

**Comment by yankinmax** [comment]
Note:
There is no `stock.inventory` model anymore in V15.
https://github.com/OCA/sale-workflow/blob/7822849f452b25c39f61e58ece1ff9e9483c2624/sale_automatic_workflow/tests/common.py#L50

**Comment by yankinmax** [comment]
@simahawk can you pls take a look?

**Review by simahawk — COMMENTED** [review]
LG. #1851 is getting merged. Pls drop the test req and squash the last commit on the migration one.

File: sale_automatic_workflow/__manifest__.py [hunk 1]
```diff
@@ -5,20 +5,20 @@
...
+    "version": "15.0.1.0.0",
     "category": "Sales Management",
     "license": "AGPL-3",
     "author": "Akretion, "
     "Camptocamp, "
     "Sodexis, "
     "Odoo Community Association (OCA)",
     "website": "https://github.com/OCA/sale-workflow",
-    "depends": ["sale_stock", "sales_team"],
+    "depends": ["sale_stock", "sales_team", "sale_delivery_state"],
```
Comment (rousseldenis) [inline sale_automatic_workflow/__manifest__.py:16]: Isn't it a little bit too much to pull this just for tests ?

Comment (simahawk) [reply sale_automatic_workflow/__manifest__.py:16]: Is not for tests, is the roadmap. Please check the diff on sale_order.py :wink: 
See discussion on https://github.com/OCA/sale-workflow/pull/1372
We don't need `sale_automatic_workflow_delivery_state` anymore.

**Review by celm1990 — CHANGES_REQUESTED** [review]
Please add `skip_sms `to context.
https://github.com/OCA/sale-workflow/blob/14.0/sale_automatic_workflow/models/stock_picking.py#L35

When module `stock_sms `is installed and sms is enabled, Odoo try show `wizard `  and avoid validate picking(If `validate_picking  `is enable on workflow). see https://github.com/odoo/odoo/blob/15.0/addons/stock_sms/models/stock_picking.py#L14


image

Run Cron job and Picking is not validate, if i try validate manually wizard is show
image

**Comment by yankinmax** [comment]
@simahawk @celm1990 @rousseldenis can we proceed here?

File: sale_automatic_workflow/models/stock_picking.py [hunk 2]
```diff
@@ -32,5 +32,5 @@ def validate_picking(self):
                 ):
                     for move_line in move.move_line_ids:
                         move_line.qty_done = move_line.product_uom_qty
-            picking.with_context(skip_immediate=True).button_validate()
+            picking.with_context(skip_immediate=True, skip_sms=True).button_validate()
```
Comment (rousseldenis) [inline sale_automatic_workflow/models/stock_picking.py:35]: Why this if people use it ?

Comment (rousseldenis) [reply sale_automatic_workflow/models/stock_picking.py:35]: Ok, didn't see past comment

**Review by rousseldenis — APPROVED** [review]

**Review by celm1990 — APPROVED** [review]

**Comment by yankinmax** [comment]
@simahawk is it ready to be approved?

**Review by simahawk — APPROVED** [review]

**Comment by johnny-longneck** [comment]
I was starting to test the sale_automatic_workflow module.
In my setup it was not creating invoices and automatic tests failed.

Is it not fully migrated, yet?

**Comment by yankinmax** [comment]
@johnny-longneck I wonder if such happens because at the moment of merge everything was ok. Maintainers wouldn't merge non-working migrations.

**Pull request merged**